src/newsimplification/SimplifyProperties.ts

```diff
@@ -9,23 +9,14 @@ import { Simplifier } from './Simplifier';
  * @param model the model to simplify properties into
  * @param simplifier the simplifier instance 
  */
-// eslint-disable-next-line sonarjs/cognitive-complexity
 export default function simplifyProperties(schema: Schema | boolean, model: CommonModel, simplifier : Simplifier) {
   if (typeof schema !== 'boolean' && schema.properties !== undefined) {
-    //Ensure object type is inferred
-    model.addToTypes('object');
+    model.addTypes('object');
 
     for (const [propertyName, propertySchema] of Object.entries(schema.properties)) {
       const propertyModels = simplifier.simplify(propertySchema);
       if (propertyModels.length > 0) {
```

Comment:
I guess that this if may be in `addProperty` it may be more DRY. I am not so sure in this case.

Reply:
I think it simplifies it using `addProperty`.